extensions/wsc/src/main/java/org/mule/extension/ws/api/exception/SoapFaultException.java

```diff
@@ -6,53 +6,37 @@
  */
 package org.mule.extension.ws.api.exception;
 
-import static java.util.Optional.ofNullable;
 import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
 import org.mule.runtime.api.exception.MuleRuntimeException;
-
-import java.util.Optional;
-
-import javax.xml.namespace.QName;
-
-import org.w3c.dom.Element;
+import org.mule.runtime.api.message.Message;
+import org.mule.runtime.core.api.exception.ErrorMessageAwareException;
 
 /**
  * Exception thrown by the Web Service Consumer when processing a SOAP fault. The exception contains the details about the fault.
  *
  * @since 4.0
  */
-public class SoapFaultException extends MuleRuntimeException {
-
-  private final QName faultCode;
-  private final QName subCode;
-  private final Element detail;
-
-  public SoapFaultException(QName faultCode,
-                            QName subCode,
-                            String message,
-                            Element detail) {
-    super(createStaticMessage(message));
-    this.faultCode = faultCode;
-    this.subCode = subCode;
-    this.detail = detail;
-  }
-
-  public SoapFaultException(QName faultCode, String message, Element detail) {
-    super(createStaticMessage(message));
-    this.faultCode = faultCode;
-    this.subCode = null;
-    this.detail = detail;
-  }
-
-  public QName getFaultCode() {
-    return faultCode;
+public class SoapFaultException extends MuleRuntimeException implements ErrorMessageAwareException {
+
+  private final SoapFault soapFault;
+
+  public SoapFaultException(org.apache.cxf.binding.soap.SoapFault cause) {
+    super(createStaticMessage(cause.getMessage()));
+    this.soapFault = new SoapFault(cause.getFaultCode(),
+                                   cause.getSubCode(),
+                                   cause.getOrCreateDetail(),
+                                   cause.getReason(),
+                                   cause.getNode(),
+                                   cause.getRole());
   }
 
-  public Optional<QName> getSubCode() {
-    return ofNullable(subCode);
+  @Override
+  public Message getErrorMessage() {
+    return Message.builder().payload(soapFault).build();
```

Comment:
isn't it better to generate this on the constructor rather than creating a new one each time?